scripts/src/models/orders.ts

```diff
@@ -310,12 +321,15 @@ export type ExternalOrderFactory = OrderFactory & {
 function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
 		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
-			data = Object.assign({}, data, { origin });
+			data = Object.assign(
+				{ nonce: Order.DEFAULT_NONCE },
+				data,
+				{ origin });
```

Comment:
OMG don't know how to read this

Reply:
what's the problem?
start with the default object with nonce, then update and override with `data` and then add the origin.